src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java

```diff
@@ -946,13 +979,18 @@ else if ( exon.getStopCodon() != null ) {
 
         boolean hasBeenClassified = false;
 
+        // Adjust the variant interval for the overlap check, specifically to properly test for the indel cases.
+        // That is, since we always have a preceding base with indels, we adjust the variant interval to remove this
+        // base so we can use the `overlaps` methods and get the correct results:
+        final SimpleInterval realVariationInterval = getBasesChangedIntervalIgnoringLeadingVcfContextBase(variant, altAllele);
+
         // Check for non-stop first:
-        if ( (exon.getStopCodon() != null) && (exon.getStopCodon().overlaps(variant)) ) {
+        if ( (exon.getStopCodon() != null) && (exon.getStopCodon().overlaps(realVariationInterval)) ) {
 
             boolean foundStop = false;
 
-            for (int i = 0; (i+3) < sequenceComparison.getAlignedCodingSequenceAlternateAllele().length(); i+=3 ){
-                final String codon = sequenceComparison.getAlignedCodingSequenceAlternateAllele().substring(i, i+3);
+            for (int i = 0; (i+AminoAcid.CODON_LENGTH-1) < sequenceComparison.getAlignedCodingSequenceAlternateAllele().length(); i+=AminoAcid.CODON_LENGTH ){
```

Comment:
Did you intend to add the -1 here? (in `i+AminoAcid.CODON_LENGTH-1`)

Reply:
Yes.

Otherwise the last codon would not be checked in the for loop because of the end condition.  Since the second argument to `String::substring` is exclusive, it is OK if `i == sequenceComparison.getAlignedCodingSequenceAlternateAllele().length()`.